src/post.ts

```diff
@@ -1,20 +1,27 @@
 import {EventPayload} from './EventPayload';
 
 /**
- * Used by the AnalyticsEventReport report() method to send an analytics event request via the
- * Beacon API.
- *
- * todo: Update this method to default to using fetch + keepalive unless not supported and
- * forceFetch is not set to true. Also update the description above.
+ * Used by the AnalyticsEventReport report() method to send an analytics event request via
+ * fetch, with the keepalive option set to true. If forceFetch is false and keepalive is not
+ * supported by the browser i.e. Firefox, the request is sent via the Beacon API.
  *
  * @param url URL that the request will be sent to
  * @param body the EventPayload object
+ * @param forceFetch if true, it will force the browser to call fetch instead of sendBeacon,
+ * regardless of whether keepAlive is supported
  */
-const post = (
+const post = async (
     url: string,
-    body: EventPayload
+    body: EventPayload,
+    forceFetch: boolean,
 ) => {
-    return navigator.sendBeacon(url, JSON.stringify(body));
+    const agent = body.browserAgent;
+    // fetch + keepalive is not supported in Firefox
+    let useBeacon = (agent && agent.browser && agent.browser.startsWith('Firefox')) || (navigator.userAgent && navigator.userAgent.includes('Firefox'));
```

Comment:
Hmn does the user setting agent in the event payload really tell us if this is on firefox?

Reply:
Most resources I found suggested using the window.navigator.userAgent to determine the browser. I just added the user set browserAgent.browser as an additional check. Do you think this is sufficient? 

I did see online that using the window.navigator.userAgent is not always reliable because the userAgent can be easily spoofed (https://stackoverflow.com/questions/9847580/how-to-detect-safari-chrome-ie-firefox-and-opera-browsers). One suggestion for checking for Firefox is checking that the typeof InstallTrigger !== 'undefined', but I saw some comments that this isn't reliable as well. I also saw suggestions to use libraries e.g. https://github.com/lancedikson/bowser.